src/tests/test_circumcision.py

```diff
@@ -0,0 +1,435 @@
+                                        & pop.data[col.CIRCUMCISED]]
+        assert circ_males.isin(new_circ_males).all()
+
+    # nobody under 10 has been circumcised
+    assert len(pop.data[(pop.data[col.AGE] < 10) & (pop.data[col.VMMC])]) == 0
+
+
+def test_vmmc_case_i1():
+
+    N = 100000
+    start_date = date(2022, 1, 1)
+    time_step = timedelta(days=90)
+
+    # build population
+    pop = Population(size=N, start_date=start_date)
+    pop.data[col.SEX] = SexType.Male
+    pop.data[col.CIRCUMCISED] = False
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+
+    # case 1
+    # circumcision stops in 10-14 year olds and
+    # increases in 15-19 year olds after year_interv
+    pop.circumcision.circ_inc_rate_year_i = 1
+    pop.circumcision.covid_disrup_affected = 0
+    pop.circumcision.vmmc_disrup_covid = 0
+    pop.circumcision.mc_int = 2008
+    pop.circumcision.year_interv = 2022
+
+    # evolve population
+    pop.data.age += time_step.days / 365
+    pop.circumcision.update_vmmc(pop)
+    # nobody under 15 has been circumcised
+    assert len(pop.data[(pop.data[col.AGE] < 15) & (pop.data[col.VMMC])]) == 0
+
+    # get stats
+    no_vmmc = len(pop.data[pop.data[col.VMMC]
+                           & (pop.data[col.AGE] >= 15)
+                           & (pop.data[col.AGE] < 20)])
+    male_population = pop.data[(pop.data[col.SEX] == SexType.Male)
+                               & (pop.data[col.AGE] >= 15)
+                               & (pop.data[col.AGE] < 20)]
+    prob_circ = ((2013 - pop.circumcision.mc_int) + (2019 - 2013)
+                 * pop.circumcision.rel_incr_circ_post_2013
+                 * pop.circumcision.circ_inc_15_19) * pop.circumcision.circ_inc_rate
+    # cap probability at 1
+    if (prob_circ > 1):
+        prob_circ = 1
+    mean = len(male_population) * prob_circ
+    stdev = sqrt(mean * (1 - prob_circ))
+    # check circumcised value is within 3 standard deviations
+    assert mean - 3 * stdev <= no_vmmc <= mean + 3 * stdev
+
+
+def test_vmmc_case_i2():
+
+    N = 100000
+    start_date = date(2021, 12, 1)
+    time_step = timedelta(days=90)
+
+    # build population
+    pop = Population(size=N, start_date=start_date)
+    pop.data[col.CIRCUMCISED] = False
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+
+    # case 2
+    # no further circumcision after year_interv
+    pop.circumcision.circ_inc_rate_year_i = 2
+    pop.circumcision.covid_disrup_affected = 0
+    pop.circumcision.vmmc_disrup_covid = 0
+    pop.circumcision.mc_int = 2008
+    pop.circumcision.year_interv = 2022
+
+    # evolve population
+    pop.data.age += time_step.days / 365
+    pop.circumcision.update_vmmc(pop)
+    circ_males = pop.data.index[(pop.data[col.SEX] == SexType.Male) & pop.data[col.CIRCUMCISED]]
+    pop.date += time_step
+    pop.data.age += time_step.days / 365
+    pop.circumcision.update_vmmc(pop)
```

Comment:
In the SAS code I think that this also affects birth circumcision, although I'm not sure if that is intentional. We should raise this with the HIV team as this would impact our birth circumcision code